### DISCUSSION TITLE
Support watchOS?

### DISCUSSION THREAD

**Discussion by plusmobileapps** [badoo/MVIKotlin#226] [Q&A]
I see in the read me that the following supported platforms are supported: 

> Android, JVM, JavaScript, iosX64, iosArm64, macosX64, linuxX64

Was wondering if there is any reason why watchOS couldn't be supported as well? Or does that add a bunch of different complexity to supporting?

**Comment by arkivanov** [comment]
There should be no blockers, it should be as easy as adding a few configuration lines. I was waiting for a use case, is there any?

**Reply by plusmobileapps** [reply]
Well the idea I have might also require some support in Decompose to support watchOS as I was wanting to use the same routing logic and store in a watch app using swiftUI as a companion to the mobile app. So I am not sure of the lift needed in the Decompose library to also support watchOS, but would be nice to be able to easily add a watch app using the same code. 

I had written a flutter app to track my pup's potty schedule and health and was thinking about writing a watch app to help log stuff, but the process of writing a watchOS app seemed tedious passing the entire app state as JSON and then writing a swift UI to reflect it. I was wanting to rewrite it in KMP anyways using MVI since I enjoy writing kotlin way more and was just curious as I start building that out if it was possible to easily write a watch app using the same code shared code from my components.

**Reply by arkivanov** [reply]
Nice use case! It will be easy to add `watchos` to Decompose as well. So if you think you need it, please file tickets and I will take care of the rest.

**Reply by plusmobileapps** [reply]
That's great to hear that it is not too difficult to add, I don't need it quite yet as I have just started rewriting the mobile apps themselves first but will file some tickets as I'm getting closer to play around with it, thanks!
